### ISSUE TITLE
how to extend listModelHandler in Lua plugin?

### ISSUE THREAD

**Issue by kambala-decapitator** [marta-file-manager/marta-issues#616]
I want to customize sorting files in a Lua plugin, so I checked API and found that `listModelHandler` seems to suit my goal. There's no example of using it, but it looks similar to `action`, so I wrote (`plugin {}` is omitted):

```lua
listModelHandler { -- <-- error here
  directoryChanged = function(actionContext)
    local activePane = actionContext.activePane
    local model = activePane.model
    local originalCompare = model.ordering.compare
    print(originalCompare)
    
    model.ordering.compare = function(first, second)
      print("comparing ", first, second)
      return originalCompare(first, second)
    end
  end
}
```
But it fails to load with error:

```
Unable to run Lua plugin from path /Users/kambala/Library/Application Support/org.yanex.marta/Plugins/package_ordering.lua: Unexpected type: luaFunction expected, got table
stack traceback:
	[C]: in ?
	[C]: in function 'listModelHandler'
	[string "package_ordering.lua"]:10: in main chunk
```

Simple `listModelHandler  {}` fails with the same error.

Then I changed `listModelHandler` to be a function instead:

```lua
listModelHandler = function(context)
  -- what to write here?
end
```

Plugin loads, but what next?

This doesn't work either:

```lua
listModelHandler.directoryChanged = function(actionContext)
  -- ...
end -- <-- error here
```

```
Unable to run Lua plugin from path /Users/kambala/Library/Application Support/org.yanex.marta/Plugins/package_ordering.lua: [string "package_ordering.lua"]:33: attempt to index a function value (global 'listModelHandler')
stack traceback:
	[C]: in metamethod '__newindex'
	[string "package_ordering.lua"]:33: in main chunk
```

**Comment by kambala-decapitator** [comment]
tried to do the same in Swift.

```swift
public class FilePackageOrderPlugin: NSObject, Plugin {
 // ...
}

public class FileListSorter: ListModelHandler {
    public func onDirectoryChanged(context: ActionContext) { print("hello world") }
    public func onModelRefreshed(context: ActionContext) {}
    public func onStateUpdated(context: ActionContext) {}
}
```

Checked in Console.app: plugin is loaded successfully, but `hello world` isn't appearing in the logs. Where should I put `FileListSorter`? Tried adding it as `public let components = [FileListSorter()]` to `FilePackageOrderPlugin`, but that didn't help.

API docs are quite poor on writing non-action plugins and no such examples as well. Besides, plugin examples haven't been updated to v10 API.

**Comment by yanex** [comment]
Hello @kambala-decapitator!

The problem with `listModelHandler` will be fixed in the next release of Marta.

```
listModelHandler {
        locationChanged = function(context)
            local parent = context.activePane.model.folder
            if not parent then return end

            local text = "Location changed: " .. parent.path
            context.activePane.view:showNotification(text, pluginId .. ".notification")
        end
    }
```

However, changing ordering from Lua is not supported yet. I will address it in 0.8.1 (#667).

**Comment by yanex** [comment]
By the way, what exact sorting do you need in Marta? It might make sense to implement it in Marta itself.

**Comment by kambala-decapitator** [comment]
I wanted to have something similar to Path Finder:

![Screenshot 2020-09-01 at 20 48 19](https://user-images.githubusercontent.com/1557784/91888149-7a561c00-ec94-11ea-9799-504bfffdc912.png)

**Comment by yanex** [comment]
Thanks! I think it definitely makes sense to support it in Marta itself. Not sure about the next version, but anyway - #668.

**Comment by yanex** [comment]
As the bug with `listModelHandler {}` is fixed in the upcoming 0.8, let me close this issue.
